swift/ql/test/query-tests/Security/CWE-611/testXMLDocumentXXE.swift

```diff
@@ -0,0 +1,86 @@
+// --- stubs ---
+
+class Data {
+    init<S>(_ elements: S) {}
+}
+
+struct URL {
+	init?(string: String) {}
+}
+
+extension String {
+	init(contentsOf: URL) {
+        let data = ""
+        self.init(data)
+    }
+}
+
+class XMLNode {
+    struct Options : OptionSet {
+        let rawValue: Int
+        static let nodeLoadExternalEntitiesAlways = XMLNode.Options(rawValue: 1 << 0)
+        static let nodeLoadExternalEntitiesNever = XMLNode.Options(rawValue: 1 << 1)
+    }
+}
+
+class XMLElement {}
+
+class XMLDocument {
+    init(contentsOf: URL, options: XMLNode.Options = []) {}
+    init(data: Data, options: XMLNode.Options = []) {}
+    init(rootElement: XMLElement?) {}
```

Comment:
This constructor doesn't receive `XMLNode.Options` to enable external entities, so it can't be a sink.